src/server/modes/base/maintenance/players/update.ts

```diff
@@ -181,8 +182,25 @@ export default class GamePlayersUpdate extends System {
    */
   onUpdatePlayers(frame: number, frameFactor: number): void {
     const now = Date.now();
+    const { afkDisconnectTimeout } = this.app.config;
 
     this.storage.playerList.forEach(player => {
+      /**
+       * Disconnect if AFK timeout configured (non-zero) and player inactivity is past that limit
+       */
+      if (
+        afkDisconnectTimeout > 0 &&
+        now - player.times.lastMove > afkDisconnectTimeout * 60 * 1000
+      ) {
+        const mainConnectionId = this.storage.playerMainConnectionList.get(player.id.current);
+        const backupConnectionId = this.storage.playerBackupConnectionList.get(player.id.current);
+
+        this.emit(ERRORS_AFK_DISCONNECT, mainConnectionId);
+        this.emit(ERRORS_AFK_DISCONNECT, backupConnectionId);
```

Comment:
I am not sure that it is necessary to duplicate the error through the backup channel. 

But forced termination of the second connection after breaking the first one is probably a good idea. Despite the fact that even if a player decides to "hold" the backup connection, it will not give him anything, maintain an unnecessary connection is redundant.

Reply:
Agreed, in https://github.com/wight-airmash/ab-server/pull/80/commits/bc42f00a7464311ca15c90355facda1651ddd07f I've changed it to emit `CONNECTIONS_BREAK` on the backup connection instead of `ERRORS_AFK_DISCONNECT`.